### PR TITLE
Basic business logic

### PR DESCRIPTION
**Pull request by gavrilin-remote**
On app launch:
- get latest block in db;
- get latest finalized block in chain;
- save events of all blocks between block in db and latest finalized(includes) in db;
- start cron job.

Start job each 6 seconds, job work:
- get latest finalized block;
- save block info in db with status OPENED;
- save all events from block into db in one session;
- change block status to CLOSED;

### CODE REVIEW

File: src/controllers/event.controller.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    const startBlock = await getLatestStoredBlock();
+    const currentBlock = await getLatestFinalizedBlock();
+    console.log({startBlock, currentBlock})
+    if(!startBlock || !currentBlock || startBlock.number === currentBlock.number){
+        return;
+    }
+    for(let i = startBlock.number + 1; i<= currentBlock.number; i++){
+        saveBlockEvents(i);
+    }
+}
```
Comment (kurotych) [inline src/controllers/event.controller.ts:81]: Could you add a short code documentation of these function?

File: src/controllers/event.controller.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+//     await api.isReady;
+//
+//     await api.rpc.chain.subscribeNewHeads(async (lastHeader) =>{
+//         console.log(`block #${lastHeader.number} has hash ${lastHeader.hash}`);
+//     });
+//
+//     api.query.system.events((events)=>{
+//
+//     });
+// }
```
[the remark this answers is not in the document]
Comment (gavrilin-remote) [reply src/controllers/event.controller.ts:95]: only for testing purpose, will remove it later

File: src/controllers/event.controller.ts (hunk 1 above)
Comment (gavrilin-remote) [reply src/controllers/event.controller.ts:81]: Yeah, will add .md file

File: src/config/common.config.ts [hunk 3]
```diff
@@ -0,0 +1,304 @@
+import * as dotenv from "dotenv";
+import path from "path";
+
+dotenv.config({ path: path.join(__dirname, '../../.env') });
+
+export default {
+    JOB_INTERVAL: 6, // seconds
+    PROVIDER_SOCKET: process.env.PROVIDER_SOCKET,
+    "CUSTOM_TYPES": {
```
[the remark this answers is not in the document]
Comment (gavrilin-remote) [reply src/config/common.config.ts:9]: I took it from FE project. I'm not sure that we need it.

File: src/controllers/event.controller.ts (hunk 1 above)
Comment (kurotych) [reply src/controllers/event.controller.ts:81]: I mean comments to code.

File: src/controllers/event.controller.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+ */
+export const fetchMissedBlocks = async () => {
+  //get latest stored and closed block from db
+  const startBlock = await getLatestStoredBlock();
+  //get current finalized block from chain
+  const currentBlock = await getLatestFinalizedBlock();
+  //check is latest block from db and chain not the same block
+  if (
+    !startBlock ||
+    !currentBlock ||
```
Comment (kurotych) [inline src/controllers/event.controller.ts:121]: Could you explain why do we need these two checks?

Comment (gavrilin-remote) [reply src/controllers/event.controller.ts:121]: !startBlock - checking is db contain any closed block. Is it make sense in this case get blocks from 0 number?
!currentBlock - checking is polkadot return any finalized block now(Chance too low)

**Review by kurotych — APPROVED** [review]

**Pull request merged**